extras/indexing/src/main/java/mvm/rya/indexing/external/tupleSet/AccumuloIndexSet.java

```diff
@@ -225,173 +273,336 @@ private void setLocalityGroups(final String tableName, final Connector conn, fin
+					}
+				}
+				// no common vars implies cross product
+				if (commonVars.size() == 0 && bs.size() != 0) {
+					crossProductBs.add(bs);
+					isCrossProd = true;
+				}
+				//get a varOrder from orders in PCJ table - use at least
+				//one common variable
+				BindingSetVariableOrder varOrder = getVarOrder(
+						commonVars.getBindingNames(),
+						constants.getBindingNames());
+
+				// update constant constraints not used in varOrder and
+				// update Bindings used to form range by removing unused
+				// variables
+				commonVars.addAll(constants);
+				if (commonVars.size() > varOrder.varOrderLen) {
+					Map<String, Value> valMap = getConstantValueMap();
+					for (String s : new HashSet<String>(varOrder.unusedVars)) {
+						if (valMap.containsKey(s)
+								&& !constantConstraints.containsKey(s)) {
+							constantConstraints.put(s, valMap.get(s));
+						}
+						commonVars.removeBinding(s);
+					}
+				}
+
+				if (containsConstantConstraints
+						&& (useColumnScan || isCrossProd)) {
+					// only one range required in event of a cross product or
+					// empty BindingSet
+					// Range will either be full table Range or determined by
+					// constant constraints
+					if (crossProductRange == EMPTY_RANGE) {
+						crossProductRange = getRange(varOrder.varOrder,
+								commonVars);
+						localityGroupOrder = prefixToOrder(varOrder.varOrder);
+					}
+				} else if (!useColumnScan && !isCrossProd) {
+					// update ranges and add BindingSet to HashJoinMap if not a
+					// cross product
+					hashJoinRanges.add(getRange(varOrder.varOrder, commonVars));
+
+					prefixLen = varOrder.varOrderLen;
+					// check if common Variable Orders are changing between
+					// BindingSets (happens in case
+					// of Optional). If common variable set length changes from
+					// BindingSet to BindingSet
+					// update the HashJoinType to be VARIABLE_JOIN_VAR.
+					if (oldPrefixLen == 0) {
+						oldPrefixLen = prefixLen;
+					} else {
+						if (oldPrefixLen != prefixLen
+								&& joinType == HashJoinType.CONSTANT_JOIN_VAR) {
+							joinType = HashJoinType.VARIABLE_JOIN_VAR;
+						}
+						oldPrefixLen = prefixLen;
+					}
+					// update max prefix len
+					if (prefixLen > maxPrefixLen) {
+						maxPrefixLen = prefixLen;
+					}
+					String key = getHashJoinKey(varOrder.varOrder, commonVars);
+					bindingSetHashMap.put(key, bs);
+				}
+
+				isCrossProd = false;
+			}
+
+			// create full Range scan iterator and set column family if empty
+			// collection or if cross product BindingSet exists and no hash join
+			// BindingSets
+			if ((useColumnScan || crossProductBs.size() > 0)
+					&& bindingSetHashMap.size() == 0) {
+				Scanner scanner = accCon.createScanner(tablename, auths);
+				// cross product with no cross product constraints here
+				scanner.setRange(crossProductRange);
+				scanner.fetchColumnFamily(new Text(localityGroupOrder));
+				return new PCJKeyToCrossProductBindingSetIterator(scanner,
```

Comment:
dumb question but how are we supporting named graphs (context) in pcjs?  This doesn't look like we are limiting to a specific context here.